keps/168-2-pending-workloads-visibility/README.md

```diff
@@ -0,0 +1,313 @@
+
+Similarly to the ```metrics-server``` the server will be implemented with the [apiserver library](https://github.com/kubernetes/apiserver), which provides authentication and authorization.
+
+All computation will be done on-demand without additional reconcile loops.
+
+The server provides that:
+- Pending workloads are returned according to their actual status without significant delay. This includes adding and removing/admitting new workloads with various priorities,
+- Adding workloads to one LocalQueues results in position changes for workloads submitted to other LocalQueues,
+- Data is consistent across all LocalQueues,
+- User with only LocalQueue visibility cannot access the list of pending workloads for ClusterQueue.
+
+### API Details
+
+We introduce a new API that will extend the existing one.
+
+There will be separate endpoints exposing the information about pending workloads for LocalQueues, and ClusterQueues. Each endpoint exposes information about a pending workload, such as:
+- workload's position in a ClusterQueue,
+- workload's position in a LocalQueue,
+- workload's priority,
+- creation timestamp.
+
+The API does not allow for the modification of any objects.
+
+Regular users will have access only to the LocalQueues they are assigned to. However, they will be able to fetch information about the global position of a workload in a ClusterQueue, without any details about workloads in different LocalQueues. 
+
+Administrators will have access to all the data at the ClusterQueue level. They will be able to view all the workloads, no matter the LocalQueues the workloads are assigned to.
+
+The API also allows user to fetch information about part of the Cluster/LocalQueue from position X to Y. There are two query parameters to do so:
+- `offset` indicates position of the first fetched workload - default: `0`
+- `limit` indicates max number of workloads to be fetched - default: `1000`
+
+Since the K8s Aggregation Layer does not support pagination out of the box, we do not want to commit to it since it would require significant effort. However, it can be implemented on the client side. We also provided above-mentioned query parameters that can help in cases where one does not want to fetch the whole Cluster/LocalQueue. Additionally, we expose [the endpoint to fetch information about a single pending workload](#fetch-information-about-a-single-pending-workload), so there is a way to query its position without listing all the remaining pending workloads.
```

Comment:
Update it, since we actually now figured out how to support pagination server side :)